### PR TITLE
initial travis.yml and dockerfile

### PR DESCRIPTION
**Pull request by LtChae**

### CODE REVIEW

**Review by sukaant — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    && apt-get -y install build-essential \
+    && apt-get -y install locales \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN pip3 install 'poetry==1.0.9'
+
+ADD pyproject.toml poetry.lock /
+
+COPY soap_to_rest /soap_to_rest
+RUN chmod +x soap_to_rest/start.sh
```
Comment (jessie-morris) [inline Dockerfile:21]: Does copy not pull in x permissions?  Also if you wanted to be pedantic you could do u+x here.

Comment (LtChae) [reply Dockerfile:21]: Not sure. A lot of this is copied from another pattern we have.

**Review by JarredOlson — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+FROM base-python as test
+COPY ./tests /tests
```
Comment (jessie-morris) [inline Dockerfile:24]: feels weird we're running the tests in the derived docker?  Is this what runs in production?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+RUN apt-get -y update \
+    && apt-get -y install libspatialindex-dev \
+    && poetry install
+RUN poetry run pytest /tests
+
+FROM base-python as production
+
+COPY nginx.conf /etc/nginx
+
+WORKDIR /soap_to_rest
```
Comment (jessie-morris) [inline Dockerfile:34]: What service is running here and will kill the docker when it dies?

File: Dockerfile (hunk 2 above)
Comment (jessie-morris) [reply Dockerfile:24]: disregard, i see the third image now, the  spacing fooled me

Comment (LtChae) [reply Dockerfile:24]: The tests are only run when you build the image, the production derived image is what gets published.

File: Dockerfile (hunk 3 above)
Comment (jessie-morris) [reply Dockerfile:34]: like "CMD ["nginx", "-g", "daemon off;"]"

Comment (LtChae) [reply Dockerfile:34]: There's no entrypoint set at the moment. Might be a good idea for usability? Could just be taken care of in whatever runs this.

File: nginx.conf [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    server {
+        listen       80 default_server;
+        listen       [::]:80 default_server;
+        server_name  localhost;
+        root         /var/www/html;
+
+        location / {
+            proxy_buffering off;
+            proxy_pass http://unix:/tmp/hypercorn.sock;
```
Comment (jessie-morris) [inline nginx.conf:34]: what is this?

**Review by jessie-morris — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (jessie-morris) [reply Dockerfile:24]: Might wanna fix that errant newline between the third from and its first command, fooled me and i lookeed over it.

File: nginx.conf (hunk 4 above)
Comment (LtChae) [reply nginx.conf:34]: I'll let @bbrewer-forge explain that I think.

**Pull request merged**